src/_langs/en/updates/2015-06-12-Media-Source-Extensions-for-Audio.markdown

```diff
@@ -0,0 +1,373 @@
+            function(data) { onAudioLoaded(data, index); });
+      } else {
+        // We've loaded all available segments, so tell MediaSource there are no
+        // more buffers which will be appended.
+        mediaSource.endOfStream();
+      }
+    });
+  }
+
+  // appendBuffer() will now use the timestamp offset and append window settings
+  // to filter and timestamp the data we're appending.
+  //
+  // Note: While this demo uses very little memory, more complex use cases need
+  // to be careful about memory usage or [garbage collection](#garbage-collection) may remove ranges of
+  // media in unexpected places.
+  sourceBuffer.appendBuffer(data);
+}
+{% endhighlight %}
+
+## A Seamless Waveform
+
+Lets see what our shiny new code has accomplished by taking another look at the waveform after we've applied our append windows. Below, you can see that the silent section at the end of _[sintel_0.mp3](sintel/sintel_0.mp3)_ (in red) and the silent section at the beginning of _[sintel_1.mp3](sintel/sintel_1.mp3)_ (in blue) have been removed; leaving us with a seamless transition between segments.
+
+<div class="waveform-container">
+
+<p style="text-align: center;">
+  <img src="/web/updates/images/2015-06-12-media-source-extensions-for-audio/mp3_mid.png" alt="Joining of sintel_0.mp3 and sintel_1.mp3" style="max-width: 60%; height: auto;">
+</p>
+
+## Conclusion
+
+With that we've stitched all five segments seamlessly into one and have subsequently reached the end of our demo. Before we go, you may have noticed that our _onAudioLoaded()_ method has no consideration for containers or codecs... Which means all of these techniques will work irrespective of the container or codec type; below you can replay the original demo DASH-ready fragmented MP4 instead of MP3.
+
+<iframe seamless style="border: 0; height: 335px; width: 100%" src="https://simpl.info/mse/audio/mp4gapless"></iframe>
+
+If you'd like to know more check the appendices below for a deeper look at gapless content creation and metadata parsing. You can also explore [_gapless.js_](gapless.js) for a closer look at the code powering this demo.
+
+Thanks for reading!
+
+## Appendix A: Creating Gapless Content
+
+Creating gapless content can be hard to get right. Below we'll walk through how the [Sintel](http://www.sintel.org/) media used in this demo were created. To start you'll need a copy of the [lossless FLAC soundtrack](http://media.xiph.org/sintel/Jan_Morgenstern-Sintel-FLAC.zip) for Sintel; for posterity, the SHA1 is included below. For tools, you'll need [FFmpeg](http://ffmpeg.org/), [MP4Box](http://gpac.wp.mines-telecom.fr/mp4box/), [LAME](http://lame.sourceforge.net/), and an OSX installation with [afconvert](https://developer.apple.com/library/mac/documentation/Darwin/Reference/Manpages/man1/afconvert.1.html).
+
+{% highlight bash %}
+unzip Jan_Morgenstern-Sintel-FLAC.zip
+sha1sum 1-Snow_Fight.flac
+# 0535ca207ccba70d538f7324916a3f1a3d550194  1-Snow_Fight.flac
```

Comment:
The pound sign needs an escape character or this looks to markdown interpreters like a heading.